Automattic/wp-calypso
# Stepper: Backbone for tracking step navigation at framework level

Related to https://github.com/Automattic/wp-calypso/issues/91755

## Proposed Changes

This sets up the backbone for handling framework tracking on the various step navigation events. 

The idea is to tackle this inductively and iteratively until a final form or abstraction surfaces. So flow by flow, event by event, until we handle all the current cases. It will keep things moving forward. I have included a temporary flow property for assisting with the migrations:

```
use__Temporary__ShouldTrackEvent?: ( event: keyof NavigationControls ) => boolean;
```

We will start with `submit`, migrate all the flows, then move to the other navigation controls. At some point in the process, we will introduce the mechanism for extending the event props, as it will surface through a need to do so.

## Why are these changes being made?


Introduce step navigation tracking at the framework level.

## Testing Instructions



- Code review
- No functional changes to flows/steps
- Confirm any known flow (e.g. `/setup/newsletter`) it continues to function and track its events as expected when navigating back/forth

## Pre-merge Checklist
